test/error.cpp

```diff
@@ -65,6 +65,15 @@ class error_test
         check(condition::assign_error, error::not_number);
         check(condition::assign_error, error::not_exact);
 
+        check(condition::pointer_parse_error, error::missing_slash);
+        check(condition::pointer_parse_error, error::invalid_escape);
+
+        check(condition::pointer_use_error, error::token_not_number);
+        check(condition::pointer_use_error, error::value_is_scalar);
+        check(condition::pointer_use_error, error::not_found);
+        check(condition::pointer_use_error, error::token_overflow);
+        check(condition::pointer_use_error, error::past_the_end);
```

Comment:
do we need this condition? what would a user do with it?

Reply:
There are two reasons finding via pointer can fail: invalid pointer string and referencing an element that doesn't exist. This condition is for the latter.